app/controllers/appointments_controller.rb

```diff
@@ -0,0 +1,70 @@
+class AppointmentsController < ApplicationController
+  before_action :authenticate_user
+  before_action :set_appointment, only: %i[show update destroy]
+
+  # GET /appointments
+  def index
+    # all current user's appointments for future dates
+    @appointments = current_user.appointments
+    @upcoming_appointments = @appointments.where('date > ?', Date.today)
+    @past_appointments = @appointments.where('date < ?', Date.today)
+    @todays_appoitments = @appointments.where('date = ?', Date.today)
+    render json: {
+      Today: @todays_appoitments,
+      Upcoming: @upcoming_appointments,
+      Past: @past_appointments
+    }
+  end
+
+  # GET /appointments/1
+  def show
+    render json: @appointment
+  end
+
+  # POST /appointments
+  def create
+    teacher_id = appointment_params[:teacher_id]
+    date = DateTime.parse(appointment_params[:date])
+    time = appointment_params[:time]
+
+    if Appointment.find_by({ teacher_id: teacher_id, date: date, time: time })
```

Comment:
We are putting too much logic in our controller. This logic of finding a duplicate appointment is more correctly done in the model.

Please remember the Fat Model - Thin Controller principle

Reply:
I have broken down this logic and refactored it to use helper methods in appointment model :+1: